### ISSUE TITLE
No nested template/string literals

### ISSUE THREAD

**Issue by haykam821** [eslint/eslint#12064]
* **Please describe what the rule should do:** This rule should prevent template literals from containing nested template literals, by warning about template literals in their placeholders. 
* **What category of rule is this? (place an "X" next to just one item)**
  * [ ] Warns about a potential error (problem)
  * [ ] Suggests an alternate way of doing something (suggestion)
  * [X] Enforces code style (layout)
  * [ ] Other (please specify:)
* **Provide 2-3 code examples that this rule will warn about:**
  * This rule will warn about:

```js
const username = "haykam";
console.log(`Hello, ${`@${username}`}!`);
```

  * This rule will not warn about:

```js
const username = "haykam";
console.log(`Hello, @${username}!`);
```
```js
const username = "haykam";
const mention = `@${username}`;
console.log(`Hello, ${mention}!`);
```

* **Why should this rule be included in ESLint (instead of a plugin)?** It fills a missing gap in template literal rules ([`template-curly-spacing`](https://eslint.org/docs/rules/template-curly-spacing), [`prefer-template`](https://eslint.org/docs/rules/template-curly-spacing), [`template-tag-spacing`](https://eslint.org/docs/rules/template-tag-spacing), [`no-template-curly-in-string`](https://eslint.org/docs/rules/no-template-curly-in-string)) and nesting rules ([`max-depth`](https://eslint.org/docs/rules/max-depth), [`max-nested-callbacks`](https://eslint.org/docs/rules/max-nested-callbacks), [`no-nested-ternary`](https://eslint.org/docs/rules/no-nested-ternary)).
* **Are you willing to submit a pull request to implement this rule?** Yes.

**Comment by kaicataldo** [comment]
Do you mind clarifying what problem this proposal is attempting to solve? We have a lot of core rules, and for maintainability reasons we have a very high bar for the inclusion of new rules. Thanks for the proposal!

**Comment by haykam821** [comment]
It would help to reduce confusion, especially when using tagged template literals like with Chalk:

```js
console.log(chalk.red`Hello, ${chalk.blue("world")}!`);
```

**Comment by platinumazure** [comment]
@haykam821 Just for my understanding, for this chalk example, what would you recommend users do instead?

```js
console.log(chalk.red`Hello, ${chalk.blue`@${username}`}!`);
```

**Comment by haykam821** [comment]
@platinumazure, I'd recommend either of these solutions:

```js
// Don't use tagged template literal for first
console.log(chalk.red("Hello, " + chalk.blue(`@${username}`) + "!"));

// Variable
const mention = chalk.blue`@${username}`;
console.log(chalk.red`Hello, ${mention}!`);
```

However, tagged template literals would be harder to autofix due to their nature. This rule could warn about them, but the scope is mostly on unnecessary nests.
